packages/terra-status-view/src/StatusView.jsx

```diff
@@ -0,0 +1,254 @@
+    variants.CUSTOM,
+  ]),
+};
+
+const defaultProps = {
+  children: [],
+  isAlignedTop: false,
+  isGlyphHidden: false,
+  message: undefined,
+  title: undefined,
+  variant: variants.ERROR,
+};
+
+const contextTypes = {
+  /* eslint-disable consistent-return */
+  intl: (context) => {
+    if (context.intl === undefined) {
+      return new Error('Please add locale prop to Base component to load translations');
+    }
+  },
+};
+
+class StatusView extends React.Component {
+
+  constructor(props, context) {
+    super(props, context);
+    this.state = {
+      showGlyph: true,
+      paddingBottom: 0,
+      paddingTop: 0,
+    };
+    this.showAndCenterItems = this.showAndCenterItems.bind(this);
+    this.setInnerNode = this.setInnerNode.bind(this);
+    this.setTitleNode = this.setTitleNode.bind(this);
+    this.setButtonsNode = this.setButtonsNode.bind(this);
+    this.setMessageNode = this.setMessageNode.bind(this);
+    this.setDividerNode = this.setDividerNode.bind(this);
+    this.setGlyphNode = this.setGlyphNode.bind(this);
+  }
+
+  componentDidMount() {
+    // call after all items have rendered in the view to calculate which items
+    // can be shown and center the items inside the view appropriately
+    this.showAndCenterItems();
+  }
+
+  setInnerNode(node) {
+    if (node === null) { return; } // Ref callbacks happen on mount and unmount, element will be null on unmount
+    this.innerNode = node;
+  }
+
+  setTitleNode(node) {
+    if (node === null) { return; }
+    this.titleNode = node;
+  }
+
+  setButtonsNode(node) {
+    if (node === null) { return; }
+    this.buttonsNode = node;
+  }
+
+  setMessageNode(node) {
+    if (node === null) { return; }
+    this.messageNode = node;
+  }
+
+  setDividerNode(node) {
+    if (node === null) { return; }
+    this.dividerNode = node;
+  }
+
+  setGlyphNode(node) {
+    if (node === null) { return; }
+    this.glyphNode = node;
+  }
+
+  /**
+   * Will calculate which items inside the status view can appear based on fixed
+   * height of its container and will update the inner padding to center the items
+   * with 40% of the remaining space at the top with 60% on the bottom.
+   */
+  showAndCenterItems() {
+    const viewHeight = this.innerNode.getBoundingClientRect().height;
+
+    let paddingTop = 0;
+    let paddingBottom = 0;
+    let showGlyph = false;
+
+    // get the content heights of the nodes that are to be shown if have content and cannot be hidden
+    let componentsHeight = this.titleNode.getBoundingClientRect().height;
+    if (this.buttonsNode) {
+      componentsHeight += this.buttonsNode.getBoundingClientRect().height;
+    }
+    if (this.messageNode) {
+      componentsHeight += this.messageNode.getBoundingClientRect().height;
+    }
+    if (this.dividerNode) {
+      componentsHeight += this.dividerNode.getBoundingClientRect().height;
+    }
+
+    if (this.glyphNode && viewHeight >= componentsHeight + this.glyphNode.getBoundingClientRect().height) {
```

Comment:
can the distribution of the padding not be handled by flexbox? I get that we need to calculate to determine if the icon will fit

Reply:
yes, we do need to determine if the icon fits and, attempting to set the padding via flexbox was leading to some over-calculated heights and content being pushed well below the bounds